test/unit/components/Context/environmentFactory.spec.js

```diff
@@ -1,33 +1,31 @@
 import environmentFactory from "../../../../src/components/Context/environmentFactory";
 
-describe("Context::getEnvironment", () => {
+describe("Context::environmentFactory", () => {
   const mywindow = {
     screen: { width: 1001, height: 1002 },
     innerWidth: 1003,
     innerHeight: 1004,
     navigator: { connection: { effectiveType: "myConnectionType" } }
   };
+  let event;
+
+  beforeEach(() => {
+    event = jasmine.createSpyObj("event", ["mergeEnvironment"]);
+  });
 
   it("works", () => {
     const date = new Date(1553550978123);
     const dateProvider = () => {
       return date;
     };
-    expect(environmentFactory(mywindow, dateProvider)()).toEqual({
-      environment: {
-        browserDetails: {
-          viewportWidth: 1003,
-          viewportHeight: 1004
-        },
-        connectionType: "myConnectionType",
-        placeContext: {
-          localTime: "2019-03-25T21:56:18.123Z",
-          // browsers don't have support for setting the timezone on a date object, because of
-          // this I cannot have hard-coded integer here because depending on the default
-          // timezone of the browser you are testing with you will get different results.
-          localTimezoneOffset: date.getTimezoneOffset()
-        }
-      }
+    environmentFactory(mywindow, dateProvider)(event);
+    expect(event.mergeEnvironment).toHaveBeenCalledWith({
+      type: "browser",
+      browserDetails: {
+        viewportWidth: 1003,
+        viewportHeight: 1004
+      },
+      connectionType: "myConnectionType"
```

Comment:
Notice that `placeContext` was moved to its own "content". This is also to match the gateway spec.

Reply:
I just reviewed the xdm schema, and you are right that placeContext is its own thing.  I'm not sure why I had it inside environment before.